### PR TITLE
Make it easier to ensure semantics in widgetTests

### PR DESCRIPTION
**Pull request by dnfield**
## Description

Today, to write a `testWidgets` test that uses semantics, you need to do:

```dart
testWidgets('Description', (WidgetTester tester) {
  final SemanticsHandle semanticsHandle = tester.ensureSemantics();
  // tests that check semantics
  semanticsHandle.dispose();
});
```

This patch makes it possible to just do:

```dart
testWidgets('Description', (WidgetTester tester) {
  // tests that check semantics
}, enableSemantics: true);
```

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_test/lib/src/widget_tester.dart [hunk 1]
```diff
@@ -84,17 +84,26 @@ void testWidgets(
   WidgetTesterCallback callback, {
   bool skip = false,
   test_package.Timeout timeout,
+  bool semanticsEnabled = false,
 }) {
+  print('$description $semanticsEnabled');
```
Comment (jonahwilliams) [inline packages/flutter_test/lib/src/widget_tester.dart]:
```suggestion
```

Comment (dnfield) [reply packages/flutter_test/lib/src/widget_tester.dart]: Oops.

File: packages/flutter_test/test/widget_tester_test.dart [hunk 2]
```diff
@@ -589,6 +589,27 @@ void main() {
...
+          ),
+        ),
+      );
+
+      final SemanticsNode node = tester.getSemantics(find.text('hello'));
+      final SemanticsData semantics = node.getSemanticsData();
+      expect(semantics.label, 'hello');
+      expect(semantics.hasAction(SemanticsAction.tap), true);
+      expect(semantics.hasFlag(SemanticsFlag.isButton), true);
+    }, semanticsEnabled: true);
```
Comment (goderbauer) [inline packages/flutter_test/test/widget_tester_test.dart]: It's unfortunate that this information is at the end of the test now... I wish the named param could be listed before the test callback.

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**